src/app/core/campaign/campaign.models.ts

```diff
@@ -3,6 +3,7 @@ export interface CampaignState {
   remoteCampaign?: Campaign;
   flights: { [id: string]: FlightState };
   availability?: { [flightZone: string]: Availability };
+  allocationPreview?: { [flightId: string]: { [zone: string]: AllocationPreview } };
```

Comment:
allocationPreview state is shaped for lookup by [flightId][zone], then allocations days are looked up by date.